components/color-picker/private/swatch-picker.jsx

```diff
@@ -92,20 +93,16 @@ class SwatchPicker extends React.Component {
 					});
 				}}
 			>
-				{this.props.swatchColors.map((color, index) => {
-					const selectedColor =
-						color === this.props.color.hex || (!isColorInSwatch && index === 0);
-					return (
-						<SwatchOption
-							color={color}
-							key={color}
-							onSelect={this.props.onSelect}
-							swatchOptionRef={this.addRef(color)}
-							workingColor={this.props.color}
-							selectedColor={selectedColor}
-						/>
-					);
-				})}
+				{this.props.swatchColors.map((color, index) => (
+					<SwatchOption
+						color={color}
+						key={color}
+						onSelect={this.props.onSelect}
+						swatchOptionRef={this.addRef(color)}
+						workingColor={this.props.color}
+						tabIndex={(this.props.color && this.props.color.hex === color) || (index === 0 && !isSelectedColorInSwatch) ? 0 : -1}
```

Comment:
tabIndex of swatch option will be 0 if it's the currently selected color (so that we can tab back right into it) or if it's the first swatch option and the color is not in the set of colors (so that we can tab into the first swatch option).